### PR TITLE
Add support for multiple OAuth2 flows

### PR DESCRIPTION
**Pull request by twoplustwoone**
OAS 3 presented a possibility of adding multiple flows per authorization scheme.

This change allows `api-security-documentation` to render all of the flows, while maintaining the way it rendered the information for RAML models.

### CODE REVIEW

File: src/ApiOauth2FlowDocument.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  /**
+   * Maps grant value to a user-friendly label
+   * @param {String} grant 
+   * @return {String}
+   */
+  _getLabelForGrant(grant) {
+    switch (grant) {
+      case 'implicit':
+        return 'Implicit';
+      case 'authorization_code':
```
Comment (jarrodek) [inline src/ApiOauth2FlowDocument.js:163]: This is missing a case for `authorizationCode` of OAS.

File: src/ApiOauth2FlowDocument.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        return 'Authorization code';
+      case 'password':
+        return 'Password';
+      case 'client_credentials':
```
Comment (jarrodek) [inline src/ApiOauth2FlowDocument.js:168]: This is missing a case for `clientCredentails` for OAS

File: src/ApiOauth2FlowDocument.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        return 'Client credentials';
+      default:
+        return grant;
+    }
+  }
+
+  render() {
+    return html`<style>${this.styles}</style>
+    <div>
```
Comment (jarrodek) [inline src/ApiOauth2FlowDocument.js]: I don't see the point of putting it into a block element. This is not React :) The element itself is a block element (via `:host` style) and it doesn't need to render another element in the tree.

**Review by jarrodek — CHANGES_REQUESTED** [review]

File: src/ApiOauth2FlowDocument.js (hunk 3 above)
Comment (twoplustwoone) [reply src/ApiOauth2FlowDocument.js]: Old habits die hard I guess :)

**Review by jarrodek — APPROVED** [review]
LGTM.

**Pull request merged**